### PR TITLE
Add AGET: Migration of consolidation to oi-userland

### PR DESCRIPTION
**Pull request by fronald**
Migration of AGET from consolidation to oi-userland. According to my responsibility documented at https://wiki.openindiana.org/oi/Migration+of+consolidations+to+oi-userland

### CODE REVIEW

File: components/web/aget/Makefile [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#
```
Comment (pyhalov) [inline components/web/aget/Makefile:1]: Please, change header to new format (look at templates/justmake.mk)

File: components/web/aget/Makefile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+COMPONENT_VERSION=	0.4.1
+COMPONENT_PROJECT_URL=	http://www.enderunix.org/aget
+COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_HASH=	\
+    sha256:d17393c7f44aab38028ae71f14b572ba1839b6e085fb2092b6ebe68bc931df4d
+COMPONENT_ARCHIVE_URL=	http://www.enderunix.org/aget/$(COMPONENT_ARCHIVE)
+COMPONENT_BUGDB=	utility/aget
+
+include $(WS_TOP)/make-rules/prep.mk
```
Comment (pyhalov) [inline components/web/aget/Makefile]: Please, use $(WS_MAKE_RULES)

File: components/web/aget/Makefile [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+COMPONENT_NAME=		aget
+COMPONENT_VERSION=	0.4.1
+COMPONENT_PROJECT_URL=	http://www.enderunix.org/aget
+COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_HASH=	\
+    sha256:d17393c7f44aab38028ae71f14b572ba1839b6e085fb2092b6ebe68bc931df4d
+COMPONENT_ARCHIVE_URL=	http://www.enderunix.org/aget/$(COMPONENT_ARCHIVE)
+COMPONENT_BUGDB=	utility/aget
```
Comment (pyhalov) [inline components/web/aget/Makefile]: We don't need COMPONENT_BUGDB, it's Oracle thing

File: components/web/aget/aget.p5m [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    value=org.opensolaris.category.2008:Applications/Internet
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license aget.license license=BSD
+
+file path=usr/bin/aget
+file path=usr/share/man/man1/aget.1
+
```
Comment (pyhalov) [inline components/web/aget/aget.p5m]: Remove empty line

File: components/web/aget/Makefile (hunk 3 above)
Comment (pyhalov) [reply components/web/aget/Makefile]: But better to move COMPONENT_FMRI, COMPONENT_LICENSE, COMPONENT_CLASSIFICATION and so on from p5m file here

File: components/web/aget/Makefile [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+COMPONENT_PREP_ACTION = ( $(CP) $(SOURCE_DIR)/Makefile.Solaris $(SOURCE_DIR)/Makefile )
+COMPONENT_INSTALL_ARGS += PREFIX=$(PROTOUSRDIR)
+
+build:		$(BUILD_32_and_64)
+
+install:	$(INSTALL_32_and_64)
+
+test:		$(TEST_32_and_64)
+
```
Comment (pyhalov) [inline components/web/aget/Makefile:36]: Why there's no REQUIRED_PACKAGES?

File: components/web/aget/Makefile [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+COMPONENT_BUGDB=	utility/aget
+
+include $(WS_TOP)/make-rules/prep.mk
+include $(WS_TOP)/make-rules/justmake.mk
+include $(WS_TOP)/make-rules/ips.mk
+
+COMPONENT_PREP_ACTION = ( $(CP) $(SOURCE_DIR)/Makefile.Solaris $(SOURCE_DIR)/Makefile )
+COMPONENT_INSTALL_ARGS += PREFIX=$(PROTOUSRDIR)
+
+build:		$(BUILD_32_and_64)
```
Comment (pyhalov) [inline components/web/aget/Makefile]: You deliver only 32-bit version of aget, so why bother building both variants?

File: components/web/aget/Makefile [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+install:	$(INSTALL_32_and_64)
+
+test:		$(TEST_32_and_64)
```
Comment (pyhalov) [inline components/web/aget/Makefile]: There are no tests: make[1]: *** No rule to make target 'check'.  Stop

File: components/web/aget/patches/01_Makefile.Solaris.patch [hunk 8]
```diff
@@ -0,0 +1,14 @@
+--- aget-0.4.1/Makefile.Solaris		2016-12-21 22:44:23.411350275 -0200
```
Comment (pyhalov) [inline components/web/aget/patches/01_Makefile.Solaris.patch:1]: Please, patch Makefile to use $(CC) and $(CFLAGS), $(LDFLAGS), passed by oi-userland build system (and check that they are actually passed).  Something like 
```
CC ?= gcc
CFLAGS  = -g -W -DSOLARIS -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 $(MYCFLAGS) 
```
and exporting CC and MYCFLAGS=$(CFLAGS) in build environment should work.
Note, that if you want to deliver only 64-bit version of aget, you don't need largefile compilation environment.

File: components/web/aget/Makefile (hunk 1 above)
Comment (fronald) [reply components/web/aget/Makefile:1]: Ok.

File: components/web/aget/Makefile (hunk 2 above)
Comment (fronald) [reply components/web/aget/Makefile]: Nice!

**Comment by fronald** [comment]
All done @pyhalov 
Thanks by review.

File: components/web/aget/Makefile [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    sha256:d17393c7f44aab38028ae71f14b572ba1839b6e085fb2092b6ebe68bc931df4d
+COMPONENT_ARCHIVE_URL=	http://www.enderunix.org/aget/$(COMPONENT_ARCHIVE)
+
+include $(WS_MAKE_RULES)/prep.mk
+include $(WS_MAKE_RULES)/justmake.mk
+include $(WS_MAKE_RULES)/ips.mk
+
+COMPONENT_PREP_ACTION = ( $(CP) $(SOURCE_DIR)/Makefile.Solaris $(SOURCE_DIR)/Makefile )
+COMPONENT_INSTALL_ARGS += PREFIX=$(PROTOUSRDIR)
+COMPONENT_BUILD_ENV += BITS=$(BITS) CC=$(CC) CFLAGS="$(CFLAGS)"
```
Comment (pyhalov) [inline components/web/aget/Makefile:31]: Better add LDFLAGS="$(LDFLAGS)" and remove BITS. 
If you do this, you can remove this part of the patch:
```
+ all: $(OBJS)
+-	$(CC) -o aget $(OBJS) $(LDFLAGS)
++	$(CC) -m$(BITS) -o aget $(OBJS) $(LDFLAGS)
```

File: components/web/aget/patches/01_Makefile.Solaris.patch [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+ 
+ all: $(OBJS)
+-	$(CC) -o aget $(OBJS) $(LDFLAGS)
++	$(CC) -m$(BITS) -o aget $(OBJS) $(LDFLAGS)
+ 
+ strip: $(all)
+ 	$(STRIP) aget
+ 	
+ install:
+-	cp -f aget /usr/local/bin/aget
```
Comment (pyhalov) [inline components/web/aget/patches/01_Makefile.Solaris.patch:22]: You still install aget in $(PREFIX)/usr/bin, both 32-bit and 64-bit version. Perhaps, you just don't need to build 32-bit one?

Comment (fronald) [reply components/web/aget/patches/01_Makefile.Solaris.patch:22]: Excuse me. I made a mistake in this commit. I will repair and send again.

Comment (pyhalov) [reply components/web/aget/patches/01_Makefile.Solaris.patch:22]: @fronald any news?

Comment (fronald) [reply components/web/aget/patches/01_Makefile.Solaris.patch:22]: Sorry about delay. I was traveling to work, I came back today to my home office.

Comment (fronald) [reply components/web/aget/patches/01_Makefile.Solaris.patch:22]: Actually the build is running for 32 and 64bit. That's right. I have nothing to correct.
Buid output:

make[1]: Entering directory '/home/fronald/Projects/oi-userland/components/web/aget/build/i86'
/usr/gcc/4.9/bin/gcc -m32 -O3  -DSOLARIS -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64   -c -o main.o main.c
/usr/gcc/4.9/bin/gcc -m32 -O3  -DSOLARIS -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64   -c -o Aget.o Aget.c
/usr/gcc/4.9/bin/gcc -m32 -O3  -DSOLARIS -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64   -c -o Misc.o Misc.c
/usr/gcc/4.9/bin/gcc -m32 -O3  -DSOLARIS -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64   -c -o Head.o Head.c
/usr/gcc/4.9/bin/gcc -m32 -O3  -DSOLARIS -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64   -c -o Signal.o Signal.c
/usr/gcc/4.9/bin/gcc -m32 -O3  -DSOLARIS -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64   -c -o Download.o Download.c
/usr/gcc/4.9/bin/gcc -m32 -O3  -DSOLARIS -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64   -c -o Resume.o Resume.c
/usr/gcc/4.9/bin/gcc -m32 -o aget main.o Aget.o Misc.o Head.o Signal.o Download.o Resume.o -lpthread -lsocket -lnsl -lresolv
make[1]: Leaving directory '/home/fronald/Projects/oi-userland/components/web/aget/build/i86'
/usr/bin/touch /home/fronald/Projects/oi-userland/components/web/aget/build/i86/.built
/bin/rm -f -r /home/fronald/Projects/oi-userland/components/web/aget/build/amd64 ; /bin/mkdir -p /home/fronald/Projects/oi-userland/components/web/aget/build/amd64
/home/fronald/Projects/oi-userland/tools/cloney /home/fronald/Projects/oi-userland/components/web/aget/aget-0.4.1 /home/fronald/Projects/oi-userland/components/web/aget/build/amd64
symlink cloning /home/fronald/Projects/oi-userland/components/web/aget/aget-0.4.1 to /home/fronald/Projects/oi-userland/components/web/aget/build/amd64
(cd /home/fronald/Projects/oi-userland/components/web/aget/build/amd64; /usr/bin/env LD_OPTIONS="-M /usr/lib/ld/map.noexstk -M /usr/lib/ld/map.noexdata -M /usr/lib/ld/map.pagealign -Bdirect -z ignore" LD_EXEC_OPTIONS="-z aslr=disable"  BITS=64 CC=/usr/gcc/4.9/bin/gcc CFLAGS="-m64 -O3 " /usr/gnu/bin/make   )
make[1]: Entering directory '/home/fronald/Projects/oi-userland/components/web/aget/build/amd64'
/usr/gcc/4.9/bin/gcc -m64 -O3  -DSOLARIS -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64   -c -o main.o main.c
/usr/gcc/4.9/bin/gcc -m64 -O3  -DSOLARIS -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64   -c -o Aget.o Aget.c
/usr/gcc/4.9/bin/gcc -m64 -O3  -DSOLARIS -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64   -c -o Misc.o Misc.c
/usr/gcc/4.9/bin/gcc -m64 -O3  -DSOLARIS -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64   -c -o Head.o Head.c
/usr/gcc/4.9/bin/gcc -m64 -O3  -DSOLARIS -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64   -c -o Signal.o Signal.c
/usr/gcc/4.9/bin/gcc -m64 -O3  -DSOLARIS -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64   -c -o Download.o Download.c
/usr/gcc/4.9/bin/gcc -m64 -O3  -DSOLARIS -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64   -c -o Resume.o Resume.c
/usr/gcc/4.9/bin/gcc -m64 -o aget main.o Aget.o Misc.o Head.o Signal.o Download.o Resume.o -lpthread -lsocket -lnsl -lresolv
make[1]: Leaving directory '/home/fronald/Projects/oi-userland/components/web/aget/build/amd64'
/usr/bin/touch /home/fronald/Projects/oi-userland/components/web/aget/build/amd64/.built
```

Comment (pyhalov) [reply components/web/aget/patches/01_Makefile.Solaris.patch:22]: Yes, but you overwrite 32-bit binary with 64-bit one on install step.

Comment (pyhalov) [reply components/web/aget/patches/01_Makefile.Solaris.patch:22]: Appropriate action here seems to be building just 64-bit version

Comment (fronald) [reply components/web/aget/patches/01_Makefile.Solaris.patch:22]: Sorry, now I understand. I'll fix it now.

**Comment by fronald** [comment]
Fixed. Only 64bit build is triggered.

**Pull request merged**